src/utils/createContextWrapper.js

```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+
+/**
+ * Creates new trigger class that injects context into overlay.
+ */
+export default function createContextWrapper(Trigger, propName) {
+  return function (contextTypes) {
+    class ContextWrapper extends React.Component {
+      getChildContext() {
+        return this.props.context;
+      }
+
+      render() {
+        // Strip injected props from below.
+        const {wrapped, ...props} = this.props;
+        delete props.context;
```

Comment:
I'm not mutating props here. I'm pulling the props from this component, removing the keys that should not be forwarded to the child element, and forwarding only the subset of the props that should be forwarded.

I initially had this written as
```js
const {wrapped, context, ...props} = this.props;
```
but this produced a lint warning on the unused `context` variable.